demo/src/main/java/io/weet/demo/services/UserService.java

```diff
@@ -49,4 +49,30 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
         return new User(user.getEmail(), user.getPassword(), authorities);
     }
 
+    //add allergen and updates it to the user object in the database
+
+    public UserModel addAllergen(String email, String allergenName){
+
+        UserModel user = userRepository.findByEmail(email); //gets the user in question
+
+        user.addAllergy(allergenName); //updates user's list of allergens by adding it
+        userRepository.save(user);
+
+        return user;
+
+    }
+
+    public UserModel deleteAllergen(String email, String allergenName){
```

Comment:
similar comment as above, there's no need to search for the user again. Just pass in the UserModel object instead of `String email`

Reply:
Similar to my comment above, I would combine these two functions into one. Instead of adding/deleting individual elements, update the whole allergens list. 

So perhaps something like `public UserModel updateAllergens(UserModel user, ArrayList<Allergen> allergens)` instead.